lib/transform-qunit-assertions.js

```diff
@@ -30,14 +30,37 @@ function isAssertionOfType(node, types) {
   return types.indexOf(node.callee.property.name) !== -1;
 }
 
-function addAssertionMessage(node, options) {
+function buildMessage(node, options) {
   var str = recast.print(node).code;
 
   if (options && options.file) {
     str += ' at ' + location(node, options.file);
   }
 
-  node.arguments.push(builders.literal(str));
+  return str;
+}
+
+function addAssertionMessage(node, message) {
+  node.arguments.push(builders.literal(message));
+}
+
+function replaceAssertionMessage(node, message) {
+  if (needsMessage(node)) {
+    addAssertionMessage(node, message);
+  } else {
+    node.arguments.pop();
+    node.arguments.push(builders.literal(message));
+  }
+}
+
+function processAssertion(node, options) {
+  var message = buildMessage(node, options);
+
+  if (options && options.completionMode === 'override') {
+    replaceAssertionMessage(node, message);
+  } else if (needsMessage(node)){
+    addAssertionMessage(node, message);
+  }
```

Comment:
Also, instead of doing `options && options.completionMode === 'override'` for every node, we could set a `var` in https://github.com/wyeworks/ember-qunit-nice-errors/blob/919eb9b60365ba65305d2372e25f50d6833b1778/lib/transform-qunit-assertions.js#L85, and just pass it in

`var overrideExistingMessage = options && options.completionMode === 'override'`

and just `function processAssertion(node, overrideExistingMessage) { ...`